### PR TITLE
Rename and hide quick_error! macro (impl_error_chain_kind)

### LINKED ISSUE #186: Rename and hide quick_error macro
Being named quick_error clobbers the same macro from the quick_error crate. Rename it to `impl_error_chain_kind` and tag it `#[doc(hidden)]`

### PR DESCRIPTION
**Pull request by AndyGauge**
Fixes #186

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -148,7 +148,7 @@
 //!     }
 //!
 //!     // Define additional `ErrorKind` variants. The syntax here is
-//!     // the same as `quick_error!`, but the `from()` and `cause()`
+//!     // the same as `impl_error_chain_kind!`, but the `from()` and `cause()`
```
Comment (Yamakaky) [inline src/lib.rs]: Not good ^^

Comment (AndyGauge) [inline src/lib.rs]: So just remove the second sentence and leave "Define additional `ErrorKind` variants."?

Comment (Yamakaky) [inline src/lib.rs]: Hum, I guess yes, if the doc is sufficiently detailed about the accepted syntax.

Comment (AndyGauge) [inline src/lib.rs]: So it is well documented in src/error_chain.rs#57 but that doc comment doesn't seem to appear on any of the generated docs.  I'm not sure how to link to this documentation.

Comment (Yamakaky) [inline src/lib.rs]: This doc appears on the generated struct, like for https://docs.rs/error-chain/0.10.0/error_chain/example_generated/struct.Error.html. I'm talking about being sure the users know what is and isn't allowed by the macro syntax.

Comment (AndyGauge) [inline src/lib.rs]: Oh, I think I realized that the implementation in the macro wasn't identical to the comment.  In fact, what is implemented inside the errors matcher of the macro is a bit above my understanding of the crate.  I've put in general documentation as my limited understanding permits, but am open to better verbiage.

**Comment by Yamakaky** [comment]
Manually merged. Thank you!

**Pull request closed without merging**